qiskit/providers/ibmq/job/ibmqjob.py

```diff
@@ -279,6 +285,173 @@ def cancel(self) -> bool:
+            new_job_tags = job.tags()  # Tags are: ['tag1', 'tag2', 'tag3']
+
+        An example of the behavior when using ``overwrite=True``::
+
+            current_job_tags = job.tags()  # Tags are: ['tag1', 'tag2']
+            job.modify_tags(['tag3'], overwrite=True)
+            new_job_tags = job.tags()  # Tags are: ['tag3']
+
+        Args:
+            tags: The new tags to add or `overwrite` for this job.
+            overwrite: If ``False`` the specified `tags` are added to the
+                job's already existing tags. If ``True``, the `tags` overwrite
+                and replace the tags currently associated with this job.
+
+        Returns:
+            ``True`` if modifying the job tags was successful, else ``False``.
+
+        Raises:
+            IBMQJobInvalidStateError: If the input job tags are invalid.
+            IBMQJobApiError: If an unexpected error occurred when communicating
+                with the server.
+        """
+        validate_job_tags(tags, IBMQJobInvalidStateError)
+
+        tags_to_update = set(tags)
+        if overwrite:
+            # Add the tags associated with a job set.
+            tags_to_update.update(
+                filter(lambda tag: re.match(RE_JOB_SET_ID_LONG, tag), self._tags))
+        else:
+            tags_to_update.update(self._tags)
+
+        with api_to_job_error():
+            response = self._api.job_update_attribute(
+                job_id=self.job_id(), attr_name='tags', attr_value=list(tags_to_update))
+
+        # Get the tags from the response and check if the update was successful.
+        updated_tags = response.get('tags', None)
+        updated_tags_set = set(updated_tags) if updated_tags else None
+        update_successful = (tags_to_update == updated_tags_set)
+
+        if update_successful:
+            self._tags = updated_tags  # Cache updated tags if successful.
+
+        return update_successful
+
+    def remove_tags(self, tags: List[str]) -> bool:
+        """Remove the specified tags from the tags associated with this job.
+
+        Note:
+            If a job was submitted through an
+            :class:`IBMQJobManager<qiskit.providers.ibmq.managed.IBMQJobManager>`
+            instance, then the job will contain a tag with the prefix ``ibmq_jobset_``.
+            These tags are necessary to retrieve a job within a
+            :class:`ManagedJobSet<qiskit.providers.ibmq.managed.ManagedJobSet>`.
+            If an attempt is made to remove one of these tags, the removal for that
+            tag will be ignored.
+
+        This method plays it safe when attempting to remove tags that are not found in the
+        job's existing tags by ignoring them. As a result, this method will return ``True``
+        when no tags are removed, or not all tags are removed, if they are not actually
+        associated with this job. The following examples give a better indication of these
+        cases.
```

Comment:
Decided to go with this logic since it made the most sense. Essentially, the idea is that attempting to remove a tag that is not present is successful, since the desired tag is not part of the job's tags anyway. Also, to make the user aware of the case, a warning is issued if a tag to remove is not found in the job's tags.

Although, it could be changed so that if none of the tags to remove are found in the job's tags, the method would return `False`. This is essentially the case for the first example below (But, instead of returning `True`, returning `False`)

However, this would open up the question:  If some tags are removed and others are not (the second example), what should be returned `True` or `False`? This is the reason why I went with the logic mentioned above.